articles/openshift/oshift-how-add-domains-proxy-whitelist.md

```diff
@@ -0,0 +1,122 @@
+services: openshift
+author: Ben Bacon
+toc_rootlink: How To
+toc_sub1: 
+toc_sub2:
+toc_sub3:
+toc_sub4:
+toc_title: Permit outbound internet access in deployments with public services network connectivity
+toc_fullpath: How To/oshift-how-add-domains-proxy-whitelist.md
+toc_mdlink: oshift-how-add-domains-proxy-whitelist.md
+---
+
+# How to permit outbound access to internet hosts in deployments with public services network connectivity
+
+## Overview
+
+This article outlines the necessary steps to whitelist domain names within Squid proxy. Squid is an enabled service within v3.11 OpenShift clusters deployed with connectivity to public sector networks (such as: HSCN, PSN-A, Janet). Whitelisted domains can be accessed on the Internet via the proxy enabling you to request external resources/data on nodes that previously only had connectivity to public sector networks.
+
+## Prerequisites
+
+To complete the steps in this guide, you must have access to a cluster which is running OpenShift Container Platform v3.11 (or newer) with connectivity to a public services network. Squid proxy is not currently enabled within environments that do not have access to a public services network.
+
+You must also have access to either:
+
+- the OpenShift Web console
+- `oc`, the OpenShift command-line client (CLI). For more information, see OpenShift's [*Get Started with the CLI*](https://docs.openshift.com/container-platform/3.11/cli_reference/get_started_cli.html)
+
+## Destination domain whitelist
+
+Squid proxy is installed as a service on the control plane load balancers. Within the Squid configuration, we have specified an ACL (Access Control List) as being a list of destination domains (present in a file on the filesystem) which herein we will refer to as the whitelist. Outbound traffic from the cluster's internal network will pass through the proxy with the destination domain being compared against the whitelist for each connection attempt; if the domain exists then outbound Internet traffic is permitted, otherwise it is denied with a HTTP 403 error.
+
+A scheduled job (which runs at 0 minutes past every hour) on the OpenShift cluster Bastion host reads a Config Map named `proxy-whitelist` within the `whitelist` project. If there are any modifications to this Config Map; the job overwrites the previous custom entries within the whitelist and triggers a reconfigure task on the Squid proxy to enable the updated domains to be accessed.
+
+### Assigning non cluster-admin users rights to edit the whitelist 
+
+By default, only users who have been assigned the cluster-admin role will be able to view/edit the `proxy-whitelist` ConfigMap within the `whitelist` project. If you wish to allow non cluster-admin users to edit this object, below are some RBAC (Role Based Access Control) commands to authorise a user to do so:
+
+>[!TIP]
+>To add multiple users to these roles, simply space delimit multiple usernames within the username variable
```

Comment:
Suggest:
To add multiple users to these roles, separate each username within the `username` variable with a space.